pkg/resources/data.go

```diff
@@ -532,33 +525,30 @@ func (d *TemplateData) NodeLocalDNSCacheEnabled() bool {
 }
 
 func (d *TemplateData) KubermaticAPIImage() string {
-	apiImageSplit := strings.Split(d.kubermaticImage, "/")
-	var registry, imageWithoutRegistry string
-	if len(apiImageSplit) != 3 {
-		registry = RegistryDocker
-		imageWithoutRegistry = strings.Join(apiImageSplit, "/")
-	} else {
-		registry = apiImageSplit[0]
-		imageWithoutRegistry = strings.Join(apiImageSplit[1:], "/")
+	return d.parseImage(d.kubermaticImage)
+}
+
+func (d *TemplateData) parseImage(image string) string {
+	named, _ := reference.ParseNormalizedNamed(image)
```

Comment:
I don't like that this error is ignored. This can lead to NPEs, but I'd expect that to be very rare since we validate the input here: 

https://github.com/kubermatic/kubermatic/blob/e2fd4a8a11f54e0e6eed4c9c5c15adb4e6c743ea/pkg/controller/operator/common/defaults.go#L543-L561

I came up with two ideas, but I don't like them:

* Change functions to return an error. We would have to change the interface and I don't like that.
* Log an error and return an empty string. I don't like it because such failure is silent.

@kron4eg do you have any opinions?

Reply:
well, since it's passed the validation, we don't have to double check errors again

Reply:
I mean, it's OK